src/shared/types/utils.ts

```diff
@@ -0,0 +1,7 @@
+export type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<
```

Comment:
I wonder if there's a simpler way to achieve the same thing? Is the purpose to enforce the presence of one extra key not in the list of passed keys? I think the name might be a bit misleading too.

Tagging @aurel-l

Reply:
Looks like this came from https://docs.microsoft.com/en-us/javascript/api/@azure/keyvault-certificates/requireatleastone?view=azure-node-latest I would trust them on that.
I don't understand what is going on inside either 😅 
Maybe comment where this is from so that we are not scared next time someone stumbles upon this file.

Reply:
My intention here is to make sure that at least one of `primaryAccession` or `id` is provided to the function (ie they can't both be optional because then they could both legally be absent).
Found it on [SO](https://stackoverflow.com/questions/40510611/typescript-interface-require-one-of-two-properties-to-exist) but found it as part of the [type-fest](https://www.npmjs.com/package/type-fest) package which is pretty popular with 53M weekly downloads. Happy to install and use from here rather than having it in our codebase. Thoughts?

Reply:
Sounds good 👍🏽 And I see that in this library there are other patterns that might be useful for us.